paasta_tools/contrib/service_shard_update.py

```diff
@@ -306,6 +318,25 @@ def main(args):
                     # Add the missing definition and write to the corresponding config
                     if args.shard_name not in config_file.keys():
                         config_file[args.shard_name] = instance_config
+
+                        # Finally, add any YAML anchors to the service shard configuration
+                        if len(args.anchors) > 0:
+                            configWithAnchor = CommentedMap(
+                                {
+                                    "<<": CommentedSeq(
+                                        [
+                                            config_file[anchor]
+                                            for anchor in args.anchors
+                                            if anchor in config_file.keys()
+                                        ]
+                                    )
+                                }
+                            )
+                            # set the formatting to single line for readability
+                            configWithAnchor["<<"].fa.set_flow_style()
+                            configWithAnchor.update(config_file[args.shard_name])
+                            config_file[args.shard_name] = configWithAnchor
```

Comment:
pythonic naming nit:
```suggestion
                            config_with_anchor = CommentedMap(
                                {
                                    "<<": CommentedSeq(
                                        [
                                            config_file[anchor]
                                            for anchor in args.anchors
                                            if anchor in config_file.keys()
                                        ]
                                    )
                                }
                            )
                            # set the formatting to single line for readability
                            config_with_anchor["<<"].fa.set_flow_style()
                            config_with_anchor.update(config_file[args.shard_name])
                            config_file[args.shard_name] = config_with_anchor
```

Reply:
whoops caught me still in js brain :p